source/docs/design/overview.md

```diff
@@ -0,0 +1,63 @@
+
+---
+
+Example. A simple Rust function with a functional spec and the generated Z3 verification condition.
+
+Rust:
+
+```
+fn small_sum(a: u64, b: u64) -> u64 {
+    requires([
+        a < 128,
+        b < 128,
+    ]);
+    ensures(|result: u64| result < 256);
+    a + b
+}
+```
+
+Z3:
+
+```
+  (declare-const result@ Int)
+  (declare-const a@ Int)
+  (declare-const b@ Int)
+  ... (3 lines with heuristics machinery and u64 type invariants) ...
+  (assert (< a@ 128))
+  (assert (< b@ 128))
+  (assert
+   (not (=>
+     (= result@ (uClip 64 (+ a@ b@))) (< result@ 256)
+  )))
+```
+
+TODO, maybe replace with something with an ADT, but in the same spirit?
+
+---
+
+From our experience, Rust's memory management story, based on linear types that prevents mutable aliased references, removes a significant challenge of other tools, like Dafny, that need to encode and manage the heap. Heap-reasoning is notoriously costly in SMT-based tools, and in our case study for Veribetrkv we were able to reduce proof burden (in terms of LOCs and verification time) by up to 30%. This advantage comes naturally in a language designed naturally around linearity, like Rust.
+
+## Borrow-checking proofs
+
+As part of recent work that's under submission, we recognized the value of tracking ownership of "ghost" variables, that do not appear in the emitted code, but aid verification. For example, in concurrent code, linear "ghost" variables permit tracking of "ghost" permissions to access non-atomic memory, they can represent a thread's knowledge of a concurrent, shared-memory protocol state, or they can represent limited aliasing domains, for example as a mechanism to encapsulate and prove non-linear data-structures that require aliased references such as `Rc`. To enable using linearity to track these separation logic assertions (TODO: terminology) Verus supports three modes for variables and functions. 
+
+**`#[spec]` mode**. The specification mode is always ghost (erased at runtime) and designed to be close to the SMT solving language: because it doesn't ever represent runtime values, code in this mode is not checked for linearity and borrow rules, giving the programmer flexibility to make assertions about values that have already been consumed, for example.
+
+**`#[proof] mode`**. Proof code is ghost code used to define lemmas and prove postconditions. In order to support linear "ghost" variables to track permissions and other singly-owned tokens that can aid verification, `#[proof]` code supports linear types and is checked by the borrow checker. It is however erased at runtime.
+
+**`#[exec] mode`**. Executable code is used for the actual implementation, and abides by simlar rules as regular Rust code. `#[proof]` and `#[spec]` are used to prove the correctness of `#[exec]` code.
```

Comment:
simlar -> similar

also you might just want to write "the same" rather than "similar"